gnosis/tabula
# Bugfix: Unable to publish edited posts

## **Descripción:**
This PR addresses the issue reported in [#262](https://github.com/gnosis/tabula/issues/262) where users were unable to publish edited articles due to an indexing loop. 

## **Changes:**
1. Refactored `ArticleHeader.tsx`:
   - Separated the logic into smaller, more manageable functions for better readability and maintainability.
   - Applied the optimistic UI approach. Once the transaction is successful, users are immediately navigated to the article visualization screen without waiting for indexing in the subgraph. This enhances the user experience as they see immediate results. If users refresh, they will see the updated data.
   - Improved error handling and state management.

2. Minor changes in `usePoster.ts`:
   - Added additional logging for better debugging and traceability.

## **How to Test:**
1. Navigate to a previously created article.
2. Edit the article.
3. Click on publish.
4. The edited article should publish/index correctly without any loops.

## **Additional Notes:**
This solution not only fixes the reported issue but also aims to improve the overall user experience by not making them wait unnecessarily.
